test/stress.jl

```diff
@@ -0,0 +1,107 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+# test for proper handling of FD exhaustion
+if Sys.isunix()
+    let ps = Pipe[]
+        ulimit_n = tryparse(Int, readchomp(`sh -c 'ulimit -n'`))
+        try
+            for i = 1:100*something(ulimit_n, 1000)
+                p = Pipe()
+                Base.link_pipe!(p)
+                push!(ps, p)
+            end
+            if ulimit_n === nothing
+                @warn "`ulimit -n` is set to unlimited, fd exhaustion cannot be tested"
+                @test_broken false
+            else
+                @test false
+            end
+        catch ex
+            isa(ex, Base.UVError) || rethrow(ex)
+            @test ex.code in (Base.UV_EMFILE, Base.UV_ENFILE)
+        finally
+            foreach(close, ps)
+        end
+    end
+end
+
+# issue 13559
+if !Sys.iswindows()
+    function test_13559()
+        fn = tempname()
+        run(`mkfifo $fn`)
+        # use subprocess to write 127 bytes to FIFO
+        writer_cmds = """
+            using Test
+            x = open($(repr(fn)), "w")
+            for i in 1:120
+                write(x, 0xaa)
+            end
+            flush(x)
+            Test.@test read(stdin, Int8) == 31
+            for i in 1:7
+                write(x, 0xaa)
+            end
+            close(x)
+        """
+        p = open(pipeline(`$(Base.julia_cmd()) --startup-file=no -e $writer_cmds`, stderr=stderr), "w")
+        # quickly read FIFO, draining it and blocking but not failing with EOFError yet
+        r = open(fn, "r")
+        # 15 proper reads
+        for i in 1:15
+            @test read(r, UInt64) === 0xaaaaaaaaaaaaaaaa
+        end
+        write(p, 0x1f)
+        # last read should throw EOFError when FIFO closes, since there are only 7 bytes (or less) available.
+        @test_throws EOFError read(r, UInt64)
+        close(r)
+        @test success(p)
+        rm(fn)
+    end
+    test_13559()
+end
+
+# issue #22566
+# issue #24037 (disabling on FreeBSD)
+if !Sys.iswindows() && !(Sys.isbsd() && !Sys.isapple())
```

Comment:
Good to enable it on FreeBSD?

Reply:
I don't know, is it? 😄 If you reenable locally it can you reproducing the freezing it was causing before?

Reply:
> If you reenable locally it can you reproducing the freezing it was causing before?

I reenabled it.... seems fine now.

Reply:
Great! Thanks for checking, I'll reenable it here.